std/format.d

```diff
@@ -2066,7 +2231,79 @@ if (is(FloatingPointTypeOf!T) && !is(T == enum) && !hasToString!(T, Char))
 
     enforceFmt(n >= 0,
         "floating point formatting failure");
-    put(w, buf[0 .. min(n, buf.length-1)]);
+
+    auto len = min(n, buf.length-1);
+    ptrdiff_t dot = buf.indexOf('.');
+    if (fs.flSeparator && dot != -1)
+    {
+        ptrdiff_t firstDigit = buf.indexOfAny("0123456789");
+        ptrdiff_t ePos = buf.indexOf('e');
+        size_t j;
+
+        ptrdiff_t firstLen = dot - firstDigit;
+
+        size_t separatorScoreCnt = firstLen / fs.separators;
+
+        size_t afterDotLen;
+        if (ePos != -1)
+        {
+            afterDotLen = ePos;
+        }
+        else
+        {
+            afterDotLen = len;
+        }
+
+        if (dot != -1)
+        {
+            for (j = dot + 1; j < afterDotLen; ++j)
+            {
+                auto realJ = (j - (dot + 1));
+                if (realJ != 0 && realJ % fs.separators == 0)
+                {
+                    ++separatorScoreCnt;
+                }
+            }
```

Comment:
P.S. `afterDotLen` seems a bit misleading, because it's actually an index to the end of the mantissa. So perhaps a better name might be `afterDotIdx`?